test/util_test.dart

```diff
@@ -155,6 +159,78 @@ void overReactExample() {}''';
         expect(stripPrivateGeneratedPrefix(r'Foo_$Bar'), r'Foo_$Bar');
       });
     });
+
+    group('hasComment()', () {
+      final content = '''
+          main() { // Comment0
+            // Comment1
+            // Comment2
+            var one;
+            var two;
+            
+            // Comment3
+          }
+        ''';
+
+      final sourceFile = SourceFile.fromString(content);
+      final astNode = parseCompilationUnit(content);
+
+      test('correctly finds a comment', () {
+        expect(hasComment(astNode, sourceFile, 'Comment0'), isTrue);
+      });
+
+      test('breaks after finding first comment', () {
+        expect(hasComment(astNode, sourceFile, 'Comment1'), isFalse);
+      });
+
+      test('only finds a comment within one line of the provided node', () {
+        expect(hasComment(astNode, sourceFile, 'Comment2'), isFalse);
+      });
+    });
+
+    group('allComments()', () {
+      final content = '''
+          main() {  
+            // Comment1
+            // Comment2
+            var one;
+            var two;
+            
+            // Comment3
+          }
+        ''';
+
+      final astNode = parseCompilationUnit(content);
+      int counter;
+
+      setUp(() {
+        counter = 0;
+      });
+
+      test('correctly iterates over all comments', () {
+        for (var comment in allComments(astNode.beginToken)) {
```

Comment:
#nit the variable `comment` is never used. Is it possible to use `allComments(astNode.beginToken).length` instead of the counter?

Reply:
💯 Yep, that's much cleaner!